src/tree/index.tsx

```diff
@@ -0,0 +1,422 @@
+import { create, tsx, diffProperty } from '@dojo/framework/core/vdom';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import { createResourceMiddleware } from '@dojo/framework/core/middleware/resources';
+import theme from '@dojo/framework/core/middleware/theme';
+import focus from '@dojo/framework/core/middleware/focus';
+import { flat } from '@dojo/framework/shim/array';
+
+import { Keys } from '../common/util';
+import Icon from '../icon';
+import Checkbox from '../checkbox';
+import { ListItem } from '../list';
+
+import * as css from '../theme/default/tree.m.css';
+import LoadingIndicator from '../loading-indicator';
+
+export interface TreeNodeOption {
+	id: string;
+	parent: string;
+	value: string;
+	hasChildren: boolean;
+}
+
+export interface TreeProperties {
+	checkable?: boolean;
+	selectable?: boolean;
+	checkedNodes?: string[];
+	expandedNodes?: string[];
+	initialChecked?: string[];
+	initialExpanded?: string[];
+	selectedNode?: string;
+	disabledNodes?: string[];
+	parentSelection?: boolean;
+	onSelect?(id: string): void;
```

Comment:
I think this should probably be `value` / `initialValue` / `onValue` to mirror other widgets and the `List`

Reply:
Hm @tomdye how would this be different than `onCheck` and `onSelect`?